test/integration/usersController.test.js

```diff
@@ -106,6 +111,65 @@ describe('Users', function () {
           expect(res.body).to.be.a('object')
           expect(res.body.message).to.equal('Users returned successfully!')
           expect(res.body.users).to.be.a('array')
+          expect(res.body.users).to.not.have.property('phone')
+          expect(res.body.users).to.not.have.property('email')
+
+          return done()
+        })
+    })
+  })
+
+  describe('GET /users/self', function () {
+    it('Should return the logged user\'s details', function (done) {
+      chai
+        .request(app)
+        .get('/users/self')
+        .set('cookie', `rds-session=${jwt}`)
+        .end((err, res) => {
+          if (err) { return done() }
+
+          expect(res).to.have.status(200)
+          expect(res.body).to.be.a('object')
+          expect(res.body).to.not.have.property('phone')
+          expect(res.body).to.not.have.property('email')
+
+          return done()
+        })
+    })
+
+    it('Should return 401 if not logged in', function (done) {
+      chai
+        .request(app)
+        .get('/users/self')
+        .end((err, res) => {
+          if (err) { return done() }
+
+          expect(res).to.have.status(401)
+          expect(res.body).to.be.an('object')
+          expect(res.body).to.eql({
+            statusCode: 401,
+            error: 'Unauthorized',
+            message: 'Unauthenticated User'
+          })
+
+          return done()
+        })
+    })
+  })
+
+  describe('GET /users/self?private=true', function () {
+    it('Should return the logged user\'s details with phone and email', function (done) {
```

Comment:
@swarajpure, can you add this test in the earlier describe block? We do not need a separate describe block for this. Its the same route.